src/test/java/com/bitmovin/api/examples/CreateEncodingWithDashAndHlsAndConditionsOnGenericS3.java

```diff
@@ -0,0 +1,352 @@
+    private static String GENERIC_S3_INPUT_SECRET_KEY = "<INSERT_YOUR_SECRETKEY>";
+    private static String GENERIC_S3_INPUT_BUCKET_NAME = "BUCKET_NAME";
+    private static String GENERIC_S3_INPUT_PATH = "<INSERT_YOUR_PATH_TO_INPUT_FILE>";
+    private static S3SignatureVersion GENERIC_S3_INPUT_SIGNATURE_VERSION = S3SignatureVersion.S3_V2;
+    private static boolean GENERIC_S3_INPUT_SSL = false;
+
+    private static String GENERIC_S3_OUTPUT_HOST = "<INSERT_YOUR_HOST>";
+    private static Integer GENERIC_S3_OUTPUT_PORT = 1234;
+    private static String GENERIC_S3_OUTPUT_ACCESSKEY = "<INSERT_YOUR_ACCESSKEY>";
+    private static String GENERIC_S3_OUTPUT_SECRET_KEY = "<INSERT_YOUR_SECRETKEY>";
+    private static String GENERIC_S3_OUTPUT_BUCKET_NAME = "BUCKET_NAME";
+    private static String OUTPUT_BASE_PATH = "path/to/your/outputs/" + new Date().getTime();
+    private static S3SignatureVersion GENERIC_S3_OUTPUT_SIGNATURE_VERSION = S3SignatureVersion.S3_V2;
+    private static boolean GENERIC_S3_OUTPUT_SSL = false;
+
+    private static BitmovinApi bitmovinApi;
+
+    @Test
+    public void testEncoding() throws IOException, BitmovinApiException, UnirestException, URISyntaxException, RestException, InterruptedException
+    {
+        bitmovinApi = new BitmovinApi(API_KEY);
+        Encoding encoding = new Encoding();
+        encoding.setName("Encoding JAVA with Generic S3 Input/Output");
+        encoding.setCloudRegion(CLOUD_REGION);
+        encoding.setEncoderVersion(ENCODER_VERSION);
+        encoding = bitmovinApi.encoding.create(encoding);
+
+        GenericS3Input input = new GenericS3Input();
+        input.setHost(GENERIC_S3_INPUT_HOST);
+        input.setPort(GENERIC_S3_INPUT_PORT);
+        input.setAccessKey(GENERIC_S3_INPUT_ACCESSKEY);
+        input.setSecretKey(GENERIC_S3_INPUT_SECRET_KEY);
+        input.setBucketName(GENERIC_S3_INPUT_BUCKET_NAME);
+        input.setSignatureVersion(GENERIC_S3_INPUT_SIGNATURE_VERSION);
+        input.setSsl(GENERIC_S3_INPUT_SSL);
+        input = bitmovinApi.input.genericS3.create(input);
+
+        GenericS3Output output = new GenericS3Output();
+        output.setHost(GENERIC_S3_OUTPUT_HOST);
+        output.setPort(GENERIC_S3_OUTPUT_PORT);
+        output.setAccessKey(GENERIC_S3_OUTPUT_ACCESSKEY);
+        output.setSecretKey(GENERIC_S3_OUTPUT_SECRET_KEY);
+        output.setBucketName(GENERIC_S3_OUTPUT_BUCKET_NAME);
+        output.setSignatureVersion(GENERIC_S3_OUTPUT_SIGNATURE_VERSION);
+        output.setSsl(GENERIC_S3_OUTPUT_SSL);
+        output = bitmovinApi.output.genericS3.create(output);
+
+        AACAudioConfig aacConfiguration = new AACAudioConfig();
+        aacConfiguration.setBitrate(128000L);
+        aacConfiguration.setRate(48000f);
+        aacConfiguration = bitmovinApi.configuration.audioAAC.create(aacConfiguration);
+
+        // Add H264 configuration for encoding
+        H264VideoConfiguration videoConfiguration240p = new H264VideoConfiguration();
```

Comment:
Please put the resolutions in a configuration array at the beginning.

Reply:
Have a look at https://github.com/bitmovin/bitmovin-java/blob/master/src/test/java/com/bitmovin/api/examples/CreateEncodingWithWatermarkFilter.java#L64